azurerm/resource_arm_ad_application.go

```diff
@@ -0,0 +1,325 @@
+package azurerm
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/Azure/azure-sdk-for-go/services/graphrbac/1.6/graphrbac"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func resourceArmAdApplication() *schema.Resource {
+	return &schema.Resource{
+		Create:        resourceArmAdApplicationCreate,
+		Read:          resourceArmAdApplicationRead,
+		Update:        resourceArmAdApplicationUpdate,
+		Delete:        resourceArmAdApplicationDelete,
+		CustomizeDiff: customizeDiffAd,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"display_name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+
+			"homepage": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+
+			"identifier_uris": {
+				Type:     schema.TypeList,
+				Optional: true,
```

Comment:
is this field user-specifyable in the portal?

Reply:
I believe this is the "App ID URI" under settings->properties. You can't specify it during creation in the portal but can modify it afterwards as long as it's globally unique. The manifest section, which allows you to edit all the application settings as a JSON document, is another way of changing it through the portal but that's probably cheating :)